File: src/SpecializedClinicApplication/ClientApp/src/models/appointment.model.ts
export class Appointment {

  constructor(id?: number, reason?: string, clientId?: number, serviceId?: number, date?: Date) {
    this.id = id;
    this.reason = reason;
    this.clientId = clientId;
    this.serviceId = serviceId;
    this.date = date;
  }

  public id?: number;
  public reason?: string;
  public clientId?: number;
  public serviceId?: number;
  public date?: Date;
  public index?: number;
}

File: src/SpecializedClinicApplication/Model/AppointmentModel.cs
using SpecializedClinicApplication.Model;

namespace SpecializedClinicApplication.Data.Models.Inventory;
public class AppointmentModel
{
    public int Id { get; set; }
    public string? Reason { get; set; }
    public int ClientId { get; set; }
    public ClientModel? Client { get; set; }
    public int ServiceId { get; set; }
    public ServiceModel? Service { get; set; }
    public DateTime Date { get; set; }
}
